strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/services/ArtifactManagementServiceImplTest.java

```diff
@@ -402,27 +431,24 @@ public void testRemoveTimestampedSnapshots(@MavenRepository(repositoryId = TRTS_
             assertEquals(1, timestampedSnapshots, "Amount of timestamped snapshots doesn't equal 1.");
 
             Path snapshotArtifactPath = getSnapshotArtifactPath(artifactVersionBasePath);
+            assertNotNull(snapshotArtifactPath);
             assertTrue(snapshotArtifactPath.toString().endsWith("-3.jar"));
 
         }
 
-        //Creating timestamped snapshot with another timestamp
+        //Creating timestamped snapshot with another timestamp (current date minus 5 days)
+        timestamp = getTimestamp(5);
 
-        Calendar cal = Calendar.getInstance();
-        cal.add(Calendar.DATE, -5);
-        String timestamp = new SimpleDateFormat("yyyyMMdd.HHmmss").format(cal.getTime());
+        artifact = setArtifactsTimestampAndRedeploy(previousDateArtifactPaths,
```

Comment:
@ptirador why does this redeploy needed? can't we just call the `removeTimestampedSnapshots` without it?

Reply:
The setting of the artifact timestamp with the current date (or relative to the current date), and the artifact redeployment with that timestamp is needed in order for the removal to work. In `removeTimestampedSnapshots` method, there's an `int` parameter called `keepPeriod`, so that if the difference of days between the current date and the artifact timestamp is greater than the `keepPeriod`, the artifact is removed.